### PR TITLE
Run Halo2 MockProver even when `POWDR_GENERATE_PROOFS=false`

### PR DESCRIPTION
**Pull request by georgwiese**
This is similar to how we always run PILCOM in the Goldilocks setting, and shouldn't be expensive!

### CODE REVIEW

**Comment by chriseth** [comment]
Did you verify the difference?

File: pipeline/src/test_util.rs [hunk 1]
```diff
@@ -207,7 +205,8 @@ pub fn test_halo2_with_backend_variant(
     let is_nightly_test = env::var("IS_NIGHTLY_TEST")
         .map(|v| v == "true")
         .unwrap_or(false);
-    if is_nightly_test {
+
+    if is_nightly_test && should_generate_proofs() {
```
Comment (chriseth) [inline pipeline/src/test_util.rs:209]: I'm actually not sure we set this for the nightly run....

Can you check?

Comment (georgwiese) [reply pipeline/src/test_util.rs:209]: Looks like it: https://github.com/powdr-labs/powdr/blob/e58ddac313963f7c4a57f2ad4760b1f54eba6791/.github/workflows/nightly-tests.yml#L9

Comment (chriseth) [reply pipeline/src/test_util.rs:209]: I mean the generate_proofs variable

Comment (georgwiese) [reply pipeline/src/test_util.rs:209]: Oh, no, that one we don't :D Let me fix it.

**Comment by georgwiese** [comment]
> Did you verify the difference?

This is already quite fast (28s) on my system on `main`: `POWDR_GENERATE_PROOFS=true PILCOM=$(pwd)/pilcom/ cargo nextest run --all --features halo2,plonky3 --cargo-profile pr-tests`

With `POWDR_GENERATE_PROOFS=false` it decreases to 25s, on `main`.

On this branch, the numbers are about the same.

**Comment by leonardoalt** [comment]
can we merge?

**Comment by georgwiese** [comment]
From my side, yes

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**